samza-core/src/main/scala/org/apache/samza/config/JobConfig.scala

```diff
@@ -213,4 +214,8 @@ class JobConfig(config: Config) extends ScalaMapConfig(config) with Logging {
   def getJMXEnabled = {
     getBoolean(JobConfig.JOB_JMX_ENABLED, true);
   }
+
+  def getSystemStreamPartitionMapperFactoryName: String = {
+    get(JobConfig.SYSTEM_STREAM_PARTITION_MAPPER_FACTORY, classOf[HashSystemStreamPartitionMapperFactory].getCanonicalName)
```

Comment:
Related to my other comment. If this SSP mapper is intended to be Kafka specific, then it should return a more generic SSP mapper factory by default.

Reply:
I think this is not kafka specific and can be extended to any custom producer partitioning function.